FileManager/Content/CIDirectory.cs

```diff
@@ -0,0 +1,137 @@
+﻿namespace FileManager.Content;
+
+/// <summary>Класс, описывающий директорию.</summary>
+internal sealed class CIDirectory : DirectoryItem
+{
+    /// <summary>Директория, описываемая текущим классом.</summary>
+    private readonly DirectoryInfo _Directory;
+
+    /// <summary>Имя директории.</summary>
+    public override string Name
+    {
+        get => _Directory.Name;
+        set
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentNullException(nameof(value));
+
+            foreach (char c in _BadSymbols)
+                if (value.Contains(c))
+                    throw new ArgumentException($"Имя файла не должно содержать символы {string.Join(' ', _BadSymbols)}");
+
+            var newName = Path.Combine(_Directory.Parent!.FullName, value);
+
+            if (File.Exists(newName) || Directory.Exists(newName))
+                throw new InvalidOperationException($"Файл {newName} уже существует!");
+
+            try
+            {
+                _Directory.MoveTo(newName);
+            }
+
+            catch (Exception ex) when (ex is System.Security.SecurityException || ex is UnauthorizedAccessException)
+            {
+                throw new UnauthorizedAccessException("Нет доступа для переименования директории!");
+            }
+            catch (DirectoryNotFoundException)
+            {
+                throw new DirectoryNotFoundException("Директория не найден!");
+            }
```

Comment:
```suggestion
            catch (DirectoryNotFoundException error)
            {
                throw new DirectoryNotFoundException("Директория не найден!", error);
            }
```
Сохраните исходное исключение в создаваемом новом - как причина возникшего вашего исключения.